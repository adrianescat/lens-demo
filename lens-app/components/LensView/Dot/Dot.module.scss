.dot {
  background-color: #fff;
  position: absolute;
  height: 20px;
  width: 20px;
  border: 2px solid #fff;
  box-shadow: white 0px 0px 0px 2px, rgb(0 0 0 / 18%) 0px 2px 10px 3px;
  border-radius: 50%;
  display: block;
  z-index: 1;
  cursor: pointer;
  transition: all 0.6s cubic-bezier(0.15, 0.9, 0.34, 0.95);

  &.active {
    background-color: #1068d4;
  }
  
  &:hover {
    transform: scale(1.3);
  }

  &::after {
    border-radius: 50%;
    content: '';
    display: block;
    position: absolute;
    top: 50%;
    left: 50%;
    height: calc(100% + 4px);
    width: calc(100% + 4px);
    transform: translateX(-50%) translateY(-50%);
    animation: pulse 2s infinite;
    opacity: 0;
    visibility: hidden;
  }

  &:hover::after {
    opacity: 1;
    visibility: visible;
  }
}

@keyframes pulse {
  0% {
  -moz-box-shadow: 0 0 0 0 rgba(16, 104, 212, 0.4);
  box-shadow: 0 0 0 0 rgba(16, 104, 212, 0.4);
  }
  70% {
    -moz-box-shadow: 0 0 0 10px rgba(16,104,212, 0);
    box-shadow: 0 0 0 10px rgba(16,104,212, 0);
  }
  100% {
    -moz-box-shadow: 0 0 0 0 rgba(16,104,212, 0);
    box-shadow: 0 0 0 0 rgba(16,104,212, 0);
  }
}