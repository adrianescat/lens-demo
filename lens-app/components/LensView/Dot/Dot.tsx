import React, { useRef, useEffect, ReactElement } from 'react'
import classnames from 'classnames';
import styles from './Dot.module.scss';

export interface Props {
  name: string;
  isActive: boolean;
  top: number;
  left: number;
  onClick: () => void;
}

export default function Dot({
  name,
  isActive,
  top,
  left,
  onClick,
}: Props): ReactElement {
  const dotRef = useRef<HTMLDivElement>(null);
  useEffect(() => {
    const node = dotRef.current

    if (isActive) {
      const coords = node.getBoundingClientRect();
      window.scroll(0, window.pageYOffset + coords.top - 300);
    }
  }, [isActive])

  return (
    <span
      ref={dotRef}
      key={name}
      data-info={name}
      className={classnames(styles.dot, isActive ? styles.active : '')}
      style={{
        top: `${top}%`,
        left: `${left}%`,
      }}
      onClick={onClick}
    />
  )
}
