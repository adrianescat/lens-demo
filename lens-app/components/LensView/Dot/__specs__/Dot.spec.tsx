import React from 'react';
import { shallow } from 'enzyme';
import Dot, { Props } from '../Dot';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = {
    name: 'dot',
    isActive: false,
    top: 90,
    left: 90,
    onClick: jest.fn(),
  }
  wrapper = shallow(<Dot {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});

describe('when is active', () => {
  beforeEach(() => {
    props = {
      name: 'dot',
      isActive: true,
      top: 90,
      left: 20,
      onClick: jest.fn(),
    }
    wrapper = shallow(<Dot {...props} />);
  });

  test('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
