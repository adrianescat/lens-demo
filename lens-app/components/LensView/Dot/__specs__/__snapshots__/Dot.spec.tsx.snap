// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<span
  className="dot"
  data-info="dot"
  key="dot"
  onClick={[MockFunction]}
  style={
    Object {
      "left": "90%",
      "top": "90%",
    }
  }
/>
`;

exports[`when is active should match snapshots 1`] = `
<span
  className="dot active"
  data-info="dot"
  key="dot"
  onClick={[MockFunction]}
  style={
    Object {
      "left": "20%",
      "top": "90%",
    }
  }
/>
`;
